sp/src/game/server/ai_basenpc.h

```diff
@@ -1320,17 +1342,16 @@ class CAI_BaseNPC : public CBaseCombatCharacter,
 #endif
 	bool InteractionCouldStart( CAI_BaseNPC *pOtherNPC, ScriptedNPCInteraction_t *pInteraction, Vector &vecOrigin, QAngle &angAngles );
 	virtual bool CanRunAScriptedNPCInteraction( bool bForced = false );
-	bool IsRunningDynamicInteraction( void ) { return (m_iInteractionState != NPCINT_NOT_RUNNING && (m_hCine != NULL)); }
-	bool IsActiveDynamicInteraction( void ) { return (m_iInteractionState == NPCINT_RUNNING_ACTIVE && (m_hCine != NULL)); }
 	ScriptedNPCInteraction_t *GetRunningDynamicInteraction( void ) { return &(m_ScriptedInteractions[m_iInteractionPlaying]); }
 	void SetInteractionCantDie( bool bCantDie ) { m_bCannotDieDuringInteraction = bCantDie; }
 	bool HasInteractionCantDie( void );
+	bool HasValidInteractionsOnCurrentEnemy( void );
+	virtual bool CanStartDynamicInteractionDuringMelee() { return false; }
```

Comment:
Question: Is this ever overwritten and does it ever return true? Or is it more for future use?

Reply:
I had to use `CanStartDynamicInteractionDuringMelee()` for some E:Z2-specific changes and I decided to bring it here for general utility purposes in case other mods use custom dynamic interaction AI.

There may be some merit to overriding this function on Combine soldiers, which I did in E:Z2's case, but I'm going to leave that for a future update when I understand the effects and consequences of soldier interactions outside of E:Z2 better.